server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileResource.java

```diff
@@ -810,11 +810,11 @@ public void updatePlanIndexes(TsFileResource another) {
   }
 
   public boolean isPlanIndexOverlap(TsFileResource another) {
-    return another.maxPlanIndex >= this.minPlanIndex && another.minPlanIndex <= this.maxPlanIndex;
+    return another.maxPlanIndex > this.minPlanIndex && another.minPlanIndex < this.maxPlanIndex;
   }
 
   public boolean isPlanRangeCovers(TsFileResource another) {
-    return this.minPlanIndex <= another.minPlanIndex && another.maxPlanIndex <= this.maxPlanIndex;
+    return this.minPlanIndex < another.minPlanIndex && another.maxPlanIndex < this.maxPlanIndex;
   }
```

Comment:
Is this really necessary? This may cause many more redundant files to be loaded if two nodes generate similar files.

Reply:
I think it's necessary. When data is migrated, each file to be loaded has to reset the plan index. which will be set as max log index in log manager, so there will be multiple files with the same plan index.